src/Admin/Filter/CategoryFilter.php

```diff
@@ -32,51 +33,43 @@ public function __construct(CategoryManagerInterface $categoryManager)
         $this->categoryManager = $categoryManager;
     }
 
-    public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data): void
+    public function filter(ProxyQueryInterface $queryBuilder, string $alias, string $field, FilterData $data): void
     {
-        if (null === $data || !\is_array($data) || !\array_key_exists('value', $data)) {
+        if (!$data->hasValue()) {
```

Comment:
```suggestion
        if (!$data->hasValue() || null === $data->getValue()) {
```
I guess something like that could be done to simplify the code.